src/js/background/background.js

```diff
@@ -87,11 +87,21 @@ async function patchMaskInfo(method = "PATCH", id, data, opts=null) {
   return await response.json();
 }
 
-async function storePremiumAvailabilityInCountry() {
+async function storeRuntimeData() {
   // If we already fetched Premium availability in the past seven days,
   // don't fetch it again.
-  const existingPremiumAvailability = (await browser.storage.local.get("premiumCountries"))?.premiumCountries;
-  if (typeof existingPremiumAvailability === "object" && existingPremiumAvailability.fetchedAt > (Date.now() - 7 * 24 * 60 * 60 * 1000)) {
+  const existingPremiumAvailability = (await browser.storage.local.get("premiumCountries")).premiumCountries;
+  const existingIntroPricingEndDate = (await browser.storage.local.get("introPricingEndDate")).introPricingEndDate;
+  
+  // If we already fetched Premium availability in the past seven days,
+  // don't fetch it again.
+  const checkingRemainingDays = Date.now() - 7 * 24 * 60 * 60 * 1000;
+  if (typeof existingPremiumAvailability === "object" && 
+      typeof existingIntroPricingEndDate === "object" && 
+      existingPremiumAvailability.fetchedAt > checkingRemainingDays &&
+      existingIntroPricingEndDate.fetchedAt > checkingRemainingDays
+      ) 
+      {
     return;
   }
```

Comment:
Added this in as per Luke's suggestion:

> check for both existingPremiumAvailability AND existingIntroPricingEndDate - and if either of them doesn't exist or is 7+ day sold, re-fetch

@Vinnl @groovecoder Could you test this again and make sure it still works?

Reply:
Just tested it by installing [this stage add-on](https://github.com/mozilla/fx-private-relay-add-on/actions/runs/2975980121), signing in, then installing [this one](https://github.com/mozilla/fx-private-relay-add-on/actions/runs/2979920257), and it appeared to work well.